frontend/src/metabase/dashboard/components/Dashboard/Dashboard.tsx

```diff
@@ -533,6 +533,7 @@ function DashboardInner(props: DashboardProps) {
       {() => (
         <DashboardStyled>
           <DashboardHeaderContainer
+            id="Dashboard-Header-Container"
```

Comment:
I could use `data-element-id` for consistency. But this fix shouldn't have anything to do with embedding SDK. This fix is only for interactive embedding because we have trouble resizing the iframe to fit the dashboard. Let's also consider other features.

1. Static embedding. We don't have to use this fix because static embedding works with [`iframe-resizer`](https://github.com/davidjbradshaw/iframe-resizer) that would automatically resize the iframe to the height of the iframe content.
2. Embedding SDK. We also don't have to rely on this fix because users will just use the component directly, as there's no iframe, SDK component parent container could just expand as much as the SDK components would take space.

Actually, another solution to this bug is to make `iframe-resizer` work on interactive embedding. But due to how different our DOM structures are between interactive embedding and static embedding. `iframe-resizer` wouldn't work well out of the box for interactive embedding. I've tried a few more configurations it offers, but couldn't make it work properly, so I rely on sending `fit` event and users can use that information to scale their iframe size properly.

Reply:
Use `data-element-id` in 438db18614343dcb1447e362d264b226af85d825.